### ISSUE TITLE
No good working in golang

### ISSUE THREAD

**Issue by RAprogramm** [SmiteshP/nvim-navic#97]
In .go files it shows only functions, so variables in functions not available

**Comment by SmiteshP** [comment]
Can you share the output by `get_location` and `get_data` and what is missing in them?

**Comment by RAprogramm** [comment]
![Screencast_00004.webm](https://user-images.githubusercontent.com/70325462/226765381-fe2fb60e-5c6d-4d0c-9f0f-76beb7ccaa45.webm)


get_location

`%#NavicIconsMethod# %*%#NavicText#(*APIserver).configureStore%*`

get_data (for e in pairs(navic.get_data()) and one more for-loop)
```
scope table: 0x419abc80
icon  
kind 6
type Method
name (*APIserver).configureStore
```

**Comment by MaiLunJiye** [comment]
Maybe the cause of this problem is the same as the https://github.com/SmiteshP/nvim-navic/issues/98.

**Comment by SmiteshP** [comment]
Yep, it might be issue with `is_available` function. @RAprogramm Can you try calling `is_available` explicitly in your winbar config for which ever plugin you are using.
```lua
function() return require("nvim-navic").is_available() end
```

**Comment by marcelbeumer** [comment]
The issue is that the Go LSP (gopls) is providing less context that for example a Lua LSP (which gives too much structure imo). So with Go you only see the function, method or struct but not deeper than one level in nvim-navic.

I think treesitter might be a better fit for this plugin that LSP. Optionally having which types to put in the breadcrumb configurable.

**Comment by SmiteshP** [comment]
@marcelbeumer Can you give example of code snippet and the information shown vs what information you want to see instead? I am not well versed in golang
